<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Schultz" />

<meta name="date" content="2023-04-23" />

<title>ESS 3500 R Handbook, Lesson 11</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ESS 3500 R Handbook</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Lessons
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RHandbook_1.html">Lesson 1: Software installation</a>
    </li>
    <li>
      <a href="RHandbook_2.html">Lesson 2: R Basics</a>
    </li>
    <li>
      <a href="RHandbook_3.html">Lesson 3: Data frames</a>
    </li>
    <li>
      <a href="RHandbook_4.html">Lesson 4: Distributions and descriptive statistics</a>
    </li>
    <li>
      <a href="RHandbook_5.html">Lesson 5: Two-sample t-test</a>
    </li>
    <li>
      <a href="RHandbook_6.html">Lesson 6: Assumptions and transformations</a>
    </li>
    <li>
      <a href="RHandbook_7.html">Lesson 7: ANOVA</a>
    </li>
    <li>
      <a href="RHandbook_8.html">Lesson 8: Correlations</a>
    </li>
    <li>
      <a href="RHandbook_9.html">Lesson 9: Regressions</a>
    </li>
    <li>
      <a href="RHandbook_10.html">Lesson 10: Multiple predictors</a>
    </li>
    <li>
      <a href="RHandbook_11.html">Lesson 11: Categorical data</a>
    </li>
  </ul>
</li>
<li>
  <a href="RHandbook_templates.html">R Templates</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ESS 3500 R Handbook, Lesson 11</h1>
<h4 class="author">Emily Schultz</h4>
<h4 class="date">2023-04-23</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#lesson-11-multiple-predictors"
id="toc-lesson-11-multiple-predictors">Lesson 11: Multiple
Predictors</a>
<ul>
<li><a href="#two-way-anovas" id="toc-two-way-anovas">11.1 Two-way
ANOVAs</a></li>
<li><a href="#two-continuous-predictors-multiple-regression"
id="toc-two-continuous-predictors-multiple-regression">11.2 Two
continuous predictors (multiple regression)</a></li>
<li><a href="#optional-practice-problems"
id="toc-optional-practice-problems">Optional practice problems</a></li>
</ul></li>
</ul>
</div>

<div id="lesson-11-multiple-predictors" class="section level2">
<h2>Lesson 11: Multiple Predictors</h2>
<p>In the previous lessons, we have worked with models that have only
one independent and one dependent variable. However, it is common to
have more than one independent variable that we would like the test.
This is often particularly true when we do observational studies, and
there are a whole suite of variable that might affect the response we
measure. In this lesson, we will cover models with multiple predictors.
We will focus on examples with only two predictors, but the methods can
easily be extended to include more than two predictors as well.</p>
<div id="two-way-anovas" class="section level3">
<h3>11.1 Two-way ANOVAs</h3>
<div id="example-1-millipedes" class="section level4">
<h4>Example 1: millipedes</h4>
<p>For our first example, we will work with two categorical predictor
variables. We will use a data set from the arthropod world this time,
testing whether levels of an amino acid, alanine, vary between different
species and different sexes of millipedes.</p>
<div id="visualizing-and-building-the-models" class="section level5">
<h5>Visualizing and building the models</h5>
<p>We’ll start off by loading and visualizing our data. Be sure your
working directory is set to the location of the millipedes data file. We
will also need to change the “species” variable to a factor.</p>
<pre class="r"><code>mill &lt;- read.csv(&quot;millipedes.csv&quot;)
mill$species &lt;- as.factor(mill$species)</code></pre>
<p>Now, let’s make a box plot to visualize our data set. This will allow
us to start looking at the patterns in the data, including possible
interactions. We will put species on the x-axis and group and color our
boxes by sex. This code includes a new <code>scale_fill_manual</code>
function that allows us to manually select the colors we want to use.
(See [A note about color in R] for more information about using colors
in R.)</p>
<pre class="r"><code>library(ggplot2)
ggplot(mill, aes(x=species, y=Alanine, fill=sex)) +
  geom_boxplot() +
  scale_fill_manual(values=c(&quot;#ce9642&quot;,&quot;#3b7c70&quot;)) +
  labs(x=&quot;Species&quot;, y=&quot;Alanine levels&quot;, fill = &quot;Sex&quot;) +
  theme_classic()</code></pre>
<p><img src="RHandbook_11_files/figure-html/mill_boxplot-1.png" width="672" /></p>
<p>Based just on the graph, what patterns do you see in the data. Does
species have an effect? What about sex? Does there appear to be an
interaction between the two variables?</p>
<p>Let’s proceed with building our models. Now that we are working with
two predictor models, we will have multiple alternative models
representing four different hypotheses: (1) only sex matters, (2) only
species matters, (3) both sex and species matter but don’t interact, (4)
both sex and species matter and they interact. The models are built in
this order after the null model below.</p>
<pre class="r"><code>mill_null &lt;- lm(Alanine ~ 1, data = mill)
mill_sex &lt;- lm(Alanine ~ sex, data = mill)
mill_species &lt;- lm(Alanine ~ species, data = mill)
mill_both &lt;- lm(Alanine ~ sex + species, data = mill)
mill_int &lt;- lm(Alanine ~ sex*species, data = mill)</code></pre>
<p>To view the output of your models, type the name of each model. Just
like when we have worked with this model before, the null model will
have just an intercept. However, the alternative models will have
multiple coefficients. Let’s take a look at a couple of them.</p>
<p>First, view the coefficients of the <code>mill_both</code> model. You
should see four coefficients. After the intercept, the first coefficient
you will see is labeled “sexm”. This is the value you would add to the
intercept to get the alanine levels in male millipedes. Then you will
see two intercepts for species. The first (species2) tells you what to
add to the intercept to get alanine leves for species 2 (for a male of
species 2, you would add both the sexm and species2 coefficients). The
final coefficient is what you would add to the intercept to get the
alanine levels for species 3.</p>
<p>Now, view the coefficients of the <code>mill_int</code> model. Now,
you will see two additional coefficients representing interactions
between the sex and species variables. Based on this model, if you had a
male of species 2, in addition to adding the sexm and species2
coefficients, you would also add the sexm:species2 coefficient, which
account for how sex interacts with being species 2. For a male of
species 3, you would similarly add the coefficient for
sexm:species3.</p>
<p>I personally find it difficult to interpret what the interaction
looks like just based on the coefficients. This is where our graphs can
come in handy. Looking at our boxplot can help us see the nature of the
interaction, if any, between our variables. Based on the graph, do you
think the two independent variables interact?</p>
</div>
<div id="testing-the-models-classical-frequentist"
class="section level5">
<h5>Testing the models: Classical Frequentist</h5>
<p>To test the models using a frequentist approach, we start with the
same <code>aov</code> function that we used for a one-way ANOVA. The
only difference is that we will include both of our independent
variables in our formula. We will set up the formula in the same way we
did for our interaction model above, because we want to test both the
main effects and the interaction affect. Run the test and look at the
output using the code below. Based on the output, we might have to run
additional follow-up tests.</p>
<pre class="r"><code>mill_2way &lt;- aov(Alanine ~ sex*species,data=mill)
summary(mill_2way)</code></pre>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## sex          1 138.72  138.72  65.679 2.04e-07 ***
## species      2  55.26   27.63  13.082  0.00031 ***
## sex:species  2   6.89    3.45   1.631  0.22331    
## Residuals   18  38.02    2.11                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>In the output, you should see a table with a row for each of the main
effects (sex and species) and a row showing the results for the
interaction between the variables (sex:species). Let’s start by looking
at the interaction effect, because that affect how we interpret the main
effects. We can see from the table that the p-value for the interaction
is greater than 0.05, so the interaction affect is not significant. This
makes things easier for us because we can interpret the main effects
based on the graph and the model coefficients, without worrying about
how they interact!</p>
<p>However, just like with a one-way ANOVA, when we have a categorical
variable with more than two groups, such a species in this example, the
ANOVA only tells us that at least one group is different. It doesn’t
tell us which specific groups are different. We therefore need to do a
post hoc test to look at those pairwise differences. We can use the same
<code>TukeyHSD</code> function that we used for post hoc tests after a
one-way ANOVA. However, because we have more than one independent
variable, we can choose to include both variables in the Tukey HSD test
or only one variable. In this case, our sex variable has only two
groups, so it is not necessary to include it in the post hoc test. We
only need to include species, as shown in the code below. (If we needed
to include both variable, we would list them both in the <code>c</code>
function: <code>which=c("species","sex")</code>. If you don’t list any
variables, it will give the results for both variables independently, as
well as their interaction.)</p>
<pre class="r"><code>TukeyHSD(mill_2way,which=c(&quot;species&quot;))</code></pre>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = Alanine ~ sex * species, data = mill)
## 
## $species
##        diff        lwr        upr     p adj
## 2-1 -3.7125 -5.5670304 -1.8579696 0.0002068
## 3-1 -2.0125 -3.8670304 -0.1579696 0.0322560
## 3-2  1.7000 -0.1545304  3.5545304 0.0756331</code></pre>
<p>You interpret this output in the same way you did following the
one-way ANOVA. Which pairs of species show significant differences from
each other?</p>
</div>
<div id="testing-the-models-likelihood" class="section level5">
<h5>Testing the models: Likelihood</h5>
<p>As we have for previous tests, we will use the <code>AIC</code>
function to compare our models, but this time we have five different
models to compare, instead of just two. We can compare all five at the
same time.</p>
<pre class="r"><code>AIC(mill_null,mill_sex,mill_species,mill_both,mill_int)</code></pre>
<pre><code>##              df       AIC
## mill_null     2 127.25979
## mill_sex      3 108.40041
## mill_species  4 124.94574
## mill_both     5  93.14672
## mill_int      7  93.14888</code></pre>
<p>Based on the output table, which model was the best? Was it
significantly better than the next best model? What do the differences
in AIC values tell you about the effects of each predictor and their
interaction?</p>
</div>
</div>
<div id="example-2-hypertension" class="section level4">
<h4>Example 2: hypertension</h4>
<p>For our second example, we will again work with two categorical
predictor variables. We will use a data set from a medical test related
to hyptertension. One common cause of hypertension is high sodium
levels, which are controlled in part by an enzyme in the kidney called
Na-K-ATPase. Researchers tested the activity of Na-K-ATPase in two
strains of lab rats: a control strain and a strain selected to
spontaneously develop hypertension. They wanted to know if Na-K-ATPase
activity varied between these two strains and, if so, what sites in the
kidney varied in their enzyme activity. The two independent variables
are the rat strain (normal or hypertensive) and the kidney site (DCT,
CCD, or OMC).</p>
<div id="visualizing-and-building-the-models-1" class="section level5">
<h5>Visualizing and building the models</h5>
<p>We’ll start off by loading and visualizing our data. Be sure your
working directory is set to the location of the kidney data file.</p>
<pre class="r"><code>kidney &lt;- read.csv(&quot;kidney.csv&quot;)</code></pre>
<p>Now, let’s make a box plot to visualize our data set. We will put
site on the x-axis and group and color our boxes by lab rat strain.</p>
<pre class="r"><code>library(ggplot2)
ggplot(kidney, aes(x=site, y=enzyme, fill=hyper)) +
  geom_boxplot() +
  scale_fill_manual(values=c(&quot;#ce9642&quot;,&quot;#3b7c70&quot;)) +
  labs(x=&quot;Kidney Site&quot;, y=&quot;Enzyme levels&quot;, fill = &quot;Strain&quot;) +
  theme_classic()</code></pre>
<p><img src="RHandbook_11_files/figure-html/kidney_boxplot-1.png" width="672" /></p>
<p>Based just on the graph, what patterns do you see in the data. Do the
strains differ in their enzyme activity? What the kidney sites? Does
there appear to be an interaction between the two variables?</p>
<p>Let’s proceed with building our models. Again, we will have multiple
alternative models representing four different hypotheses: (1) only
strain matters, (2) only site matters, (3) both strain and site matter
but don’t interact, (4) both strain and site matter and they interact.
The models are built in this order after the null model below.</p>
<pre class="r"><code>kidney_null &lt;- lm(enzyme ~ 1, data = kidney)
kidney_hyper &lt;- lm(enzyme ~ hyper, data = kidney)
kidney_site &lt;- lm(enzyme ~ site, data = kidney)
kidney_both &lt;- lm(enzyme ~ hyper + site, data = kidney)
kidney_int &lt;- lm(enzyme ~ hyper*site, data = kidney)</code></pre>
<p>View the output of your models by typing the name of each model.
Based on the model outputs and the graph, without running the tests, how
would you describe the patterns in your results.</p>
</div>
<div id="testing-the-models-classical-frequentist-1"
class="section level5">
<h5>Testing the models: Classical Frequentist</h5>
<p>As in the previous example, we will start by using the
<code>aov</code> function to run the overall two-way ANOVA.</p>
<pre class="r"><code>kidney_2way &lt;- aov(enzyme ~ hyper*site, data=kidney)
summary(kidney_2way)</code></pre>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## hyper        1    459     459   7.139   0.0156 *  
## site         2   8287    4144  64.393 6.27e-09 ***
## hyper:site   2    496     248   3.854   0.0404 *  
## Residuals   18   1158      64                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>In the output, you should see a similar table as you saw in the
millipede example. Let’s start by looking at the interaction effect
again. We can see from the table that the p-value for the interaction is
less than 0.05, so the interaction affect is significant this time. This
makes things trickier because now we can’t evaluate each main effect
independently. Because the two variables interact, the each main effect
depends on the value of the other independent variable.</p>
<p>One way to deal with this non-independence is to split up our data
into groups and run multiple one-way ANOVAs (or t-tests, depending on
how many categories we have. For example, in this case, we are primarily
interested in the difference between normal and hypertensive rats.
Therefore, we can split our data into three groups, one for each kidney
site, and then run three tests to compare normal and hypertensive rats
at each site.</p>
<p>(Note: We do not want to split our data like this every time we run a
two-way ANOVA. We only use this approach when there is a significant
interaction.)</p>
<p>A downside of this approach is that we are now running multiple tests
on the same data set. We can’t use a Tukey HSD test in this particular
scenario, but what is another strategy we discussed in the last class
that could work here?</p>
<p>Now we will run t-tests to compare the normal and hypertensive rats
at each individual kidney site. We will take advantage of two new
functions: the <code>subset</code> function and the <code>with</code>
function. The <code>subset</code> function will allow us to take a
subset of the data containing only the kidney site we want. The
<code>with</code> function will tell R to apply the t.test to the subset
of the data that we took. We include the code for the t-test as the
second argument of the <code>with</code> function.</p>
<pre class="r"><code>with(subset(kidney,site==&quot;CCD&quot;),t.test(enzyme~hyper))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  enzyme by hyper
## t = -0.85682, df = 5.5936, p-value = 0.4267
## alternative hypothesis: true difference in means between group hyp and group norm is not equal to 0
## 95 percent confidence interval:
##  -26.37137  12.87137
## sample estimates:
##  mean in group hyp mean in group norm 
##              18.25              25.00</code></pre>
<p>Now try adapting this code to run the t-test for the other two sites
(DCT and OMCD). Based on the output of the three tests, which kidney
sites show differences in enzyme activity between the two strains of
mice? Don’t forget to adjust your threshold p-value based on the number
of tests you ran.</p>
</div>
<div id="testing-the-models-likelihood-1" class="section level5">
<h5>Testing the models: Likelihood</h5>
<p>Now we will use the likelihood approach and <code>AIC</code> function
to compare our models, just like we did with the millipede models.</p>
<pre class="r"><code>AIC(kidney_null,kidney_hyper,kidney_site,kidney_both,kidney_int)</code></pre>
<pre><code>##              df      AIC
## kidney_null   2 217.8267
## kidney_hyper  3 218.7425
## kidney_site   4 183.5836
## kidney_both   5 179.7022
## kidney_int    7 175.1477</code></pre>
<p>Based on the output table, which model was the best? Was it
significantly better than the next best model? What do the differences
in AIC values tell you about the effects of each predictor and their
interaction?</p>
</div>
</div>
</div>
<div id="two-continuous-predictors-multiple-regression"
class="section level3">
<h3>11.2 Two continuous predictors (multiple regression)</h3>
<p>For our next example, we will work with two continuous predictor
variable, sometimes called a multiple regression. We will use the carbon
storage data set that we worked with once before, but this time will
will look at some additional variables and consider how the levels of
active bacteria and active fungi in the soil affect total carbon
storage.</p>
<div id="visualizing-and-building-the-models-2" class="section level4">
<h4>Visualizing and building the models</h4>
<p>First, load the data.</p>
<pre class="r"><code>carbon &lt;- read.csv(&quot;carbon.csv&quot;)</code></pre>
<p>There are many variables in the data set, but we will just focus on a
few. “Tc” will be our response variable and “Active_Bacteria” and
“Active_Lifespan” will be our two predictor variables.</p>
<p>Now let’s graph our data. It is challenging to visualize two
continuous predictor variable at the same time (I think 3D graphs are
hard to read). One option is just to make separate scatter plots for
each predictor, but that does not allow you to see interactions between
the two variables. What I often do is convert one of the predictors to a
categorical variable (just for the graph, not for the test), and graph
both of the predictors at the same time. We’ll try that.</p>
<p>First, we will use the <code>mutate</code> function to create a new
variable that converts active fungi to a category. We will just use two
categories: high and low. High active fungi will be anything above the
median active fungi from our data set, and low weight will be anything
less than or equal to the median. To to this, we will use the
<code>ifelse</code> function. This function allows us to set a value for
something <strong>if</strong> it matches certain criteria (in this case,
if the active fungi is greater than the median) and set a different
values if it does not match the criteria.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
## ✔ tibble  3.1.8      ✔ dplyr   1.0.10
## ✔ tidyr   1.3.0      ✔ stringr 1.5.0 
## ✔ readr   2.1.3      ✔ forcats 1.0.0 
## ✔ purrr   1.0.1      
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>med &lt;- median(carbon$Active_Fungi)
carbon &lt;- mutate(carbon,Fungi_cat = ifelse(Active_Fungi &gt; med, &quot;High&quot;, &quot;Low&quot;))</code></pre>
<p>If you view the data set, you will now see a new variable called
“Weight_cat” that we will use to graph our data. Let’s make that graph
now. We will use “Lifespan” as our x variable, and we will use two
different colors to represent our two weight categories.</p>
<pre class="r"><code>ggplot(data=carbon, aes(x=Active_Bacteria,y=Tc,color=Fungi_cat)) +
  geom_point() +
  geom_smooth(method=&quot;lm&quot;) +
  scale_color_manual(values=c(&quot;#ce9642&quot;,&quot;#3b7c70&quot;)) +
  labs(x=&quot;Active bacteria&quot;, y=&quot;Total carbon storage&quot;,color=&quot;Active fungi&quot;) +
  theme_classic()</code></pre>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="RHandbook_11_files/figure-html/carbon_scat-1.png" width="672" /></p>
<p>What are your initial interpretations based on the graph?</p>
<p>Now, let’s build our models. Just like our first example, we will
have multiple alternative models that we test.</p>
<pre class="r"><code>carbon_null &lt;- lm(Tc ~ 1, data = carbon)
carbon_bac &lt;- lm(Tc ~ Active_Bacteria, data = carbon)
carbon_fungi &lt;- lm(Tc ~ Active_Fungi, data = carbon)
carbon_both &lt;- lm(Tc ~ Active_Bacteria + Active_Fungi, data = carbon)
carbon_int &lt;- lm(Tc ~ Active_Bacteria*Active_Fungi, data = carbon)</code></pre>
<p>Look at the summaries for some of your models. Do the coefficients
for the different variable make sense based on what you saw in your
graph?</p>
<p>We will again check our assumption of normality before testing the
models.</p>
<pre class="r"><code>resid_null_carbon &lt;- resid(carbon_null)
resid_bac_carbon &lt;- resid(carbon_bac)
resid_fungi_carbon &lt;- resid(carbon_fungi)
resid_both_carbon &lt;- resid(carbon_both)
resid_int_carbon &lt;- resid(carbon_int)</code></pre>
<pre class="r"><code>qqnorm(resid_null_carbon)
qqline(resid_null_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/carbon_qq-1.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_bac_carbon)
qqline(resid_bac_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/carbon_qq-2.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_fungi_carbon)
qqline(resid_fungi_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/carbon_qq-3.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_both_carbon)
qqline(resid_both_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/carbon_qq-4.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_int_carbon)
qqline(resid_int_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/carbon_qq-5.png" width="672" /></p>
<p>Yikes, this time, it looks like we have some skew in our data based
on the curved shape of the points in our qqplots. Before we run the
tests, we’ll try a log transform to see if that makes our residuals less
skewed. We can actually do this transform right in the formula of our
linear model, instead of adding a new variable. We will just add the
<code>log</code> function to our dependent variable (Tc).</p>
<pre class="r"><code>carbon_null &lt;- lm(log(Tc) ~ 1, data = carbon)
carbon_bac &lt;- lm(log(Tc) ~ Active_Bacteria, data = carbon)
carbon_fungi &lt;- lm(log(Tc) ~ Active_Fungi, data = carbon)
carbon_both &lt;- lm(log(Tc) ~ Active_Bacteria + Active_Fungi, data = carbon)
carbon_int &lt;- lm(log(Tc) ~ Active_Bacteria*Active_Fungi, data = carbon)</code></pre>
<p>Look at the summaries for some of your models. Do the coefficients
for the different variable make sense based on what you saw in your
graph?</p>
<p>Now, we will check the residuals again, using the models with the log
transformed data.</p>
<pre class="r"><code>resid_null_carbon &lt;- resid(carbon_null)
resid_bac_carbon &lt;- resid(carbon_bac)
resid_fungi_carbon &lt;- resid(carbon_fungi)
resid_both_carbon &lt;- resid(carbon_both)
resid_int_carbon &lt;- resid(carbon_int)</code></pre>
<pre class="r"><code>qqnorm(resid_null_carbon)
qqline(resid_null_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/logcarbon_qq-1.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_bac_carbon)
qqline(resid_bac_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/logcarbon_qq-2.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_fungi_carbon)
qqline(resid_fungi_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/logcarbon_qq-3.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_both_carbon)
qqline(resid_both_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/logcarbon_qq-4.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_int_carbon)
qqline(resid_int_carbon)</code></pre>
<p><img src="RHandbook_11_files/figure-html/logcarbon_qq-5.png" width="672" /></p>
<p>These look better now, so we’ll move on with testing our models.</p>
</div>
<div id="testing-the-models" class="section level4">
<h4>Testing the models</h4>
<p>To test the models, we will use the same approach that we used with
the first example.</p>
<pre class="r"><code>AIC(carbon_null,carbon_bac,carbon_fungi,carbon_both,carbon_int)</code></pre>
<pre><code>##              df      AIC
## carbon_null   2 177.1537
## carbon_bac    3 154.9944
## carbon_fungi  3 167.1470
## carbon_both   4 154.3212
## carbon_int    5 145.0235</code></pre>
<p>Based on the AIC models, which model is the best? Is it significantly
better than the others? Given this, along with the coefficients from the
best model and the patterns you see in your graph, what can you conclude
about the effects of the two predictor variables (bacteria and fungi
levels)?</p>
<p>### 11.3 Continuous and categorial predictors For our final example,
we will look at a blend of the previous two types of predictors and
consider a case where we have one continuous and one categorical
predictor. We will use an example of butterfly endurance and how it is
affected by both temperature and the genotype of the butterfly.</p>
</div>
<div id="visualizing-and-building-the-models-3" class="section level4">
<h4>Visualizing and building the models</h4>
<pre class="r"><code>butterfly &lt;- read.csv(&quot;butterfly.csv&quot;)</code></pre>
<pre class="r"><code>ggplot(data=butterfly, aes(x=Temp,y=Endurance,color=Genotype)) +
  geom_point() +
  geom_smooth(method=&quot;lm&quot;) +
  scale_color_manual(values=c(&quot;#ce9642&quot;,&quot;#3b7c70&quot;,&quot;#3b3a3e&quot;)) +
  labs(x=&quot;Temperature&quot;) +
  theme_classic()</code></pre>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="RHandbook_11_files/figure-html/butterfly_scat-1.png" width="672" /></p>
<p>What is your initial interpretation for how temperature and genotype
affect the endurance of butterflies, based on the graph? Does unequal
variances seem to be a problem?</p>
<p>We will now move on to building our models: one null model and our
four different alternative models.</p>
<pre class="r"><code>butter_null &lt;- lm(Endurance ~ 1, data=butterfly)
butter_temp &lt;- lm(Endurance ~ Temp, data=butterfly)
butter_geno &lt;- lm(Endurance ~ Genotype, data=butterfly)
butter_both &lt;- lm(Endurance ~ Temp + Genotype, data=butterfly)
butter_int &lt;- lm(Endurance ~ Temp*Genotype, data=butterfly)</code></pre>
<p>Before testing these models, we’ll check our residuals for
normality.</p>
<pre class="r"><code>resid_null_butter &lt;- resid(butter_null)
resid_temp_butter &lt;- resid(butter_temp)
resid_geno_butter &lt;- resid(butter_geno)
resid_both_butter &lt;- resid(butter_both)
resid_int_butter &lt;- resid(butter_int)</code></pre>
<pre class="r"><code>qqnorm(resid_null_butter)
qqline(resid_null_butter)</code></pre>
<p><img src="RHandbook_11_files/figure-html/butterfly_qq-1.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_temp_butter)
qqline(resid_temp_butter)</code></pre>
<p><img src="RHandbook_11_files/figure-html/butterfly_qq-2.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_geno_butter)
qqline(resid_geno_butter)</code></pre>
<p><img src="RHandbook_11_files/figure-html/butterfly_qq-3.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_both_butter)
qqline(resid_both_butter)</code></pre>
<p><img src="RHandbook_11_files/figure-html/butterfly_qq-4.png" width="672" /></p>
<pre class="r"><code>qqnorm(resid_int_butter)
qqline(resid_int_butter)</code></pre>
<p><img src="RHandbook_11_files/figure-html/butterfly_qq-5.png" width="672" /></p>
<p>Once again, the residuals don’t look quite normal, but there doesn’t
seem to be a lot of skew, so we’ll move on with testing our models.</p>
</div>
<div id="testing-the-models-1" class="section level4">
<h4>Testing the models</h4>
<p>Using the likelihood approach, we can again just use the
<code>AIC</code> function to compare our five models:</p>
<pre class="r"><code>AIC(butter_null,butter_temp,butter_geno,butter_both,butter_int)</code></pre>
<pre><code>##             df       AIC
## butter_null  2 103.12351
## butter_temp  3  73.47012
## butter_geno  4 105.64924
## butter_both  5  55.27518
## butter_int   7  58.88042</code></pre>
<p>Based on these AIC values, what can you conclude about the effects of
temperature, genotype, and their interaction on butterfly endurance?</p>
</div>
</div>
<div id="optional-practice-problems" class="section level3">
<h3>Optional practice problems</h3>
<p>Download the algae.csv file from Canvas. This data set contains data
on the growth of algae under 4 different light levels (low, med low, med
high, and high) and three different wave intensity (low, med, high).
Researchers want to know if the light level and wave intensity affect
algae growth. For the algae growth variable, we will be working with the
natural log of growth (log_growth).</p>
<ol style="list-style-type: decimal">
<li><p>Write the null and alternative hypotheses for the main effects
and the interaction effect.</p></li>
<li><p>Make a box plot to visualize the effect of both independent
variables (light and wave) on algae growth.</p></li>
<li><p>Run a two-way ANOVA using the classical frequentist approach. Run
any follow-up tests as necessary.</p></li>
<li><p>Build and compare your models using the likelihood approach.
Don’t forget to build models for both predictors individually, the
predictors together, and the predictor together plus their
interaction.</p></li>
<li><p>State your conclusions from your analysis, including all
necessary statistics.</p></li>
</ol>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
